src/sensor.h

```diff
@@ -236,6 +237,11 @@ namespace librealsense
         bool is_streaming() const override;
         bool is_opened() const override;
 
+        virtual rsutils::subscription register_options_value_changed_callback(
+            std::function< void( const std::map< rs2_option, std::shared_ptr< option > > & ) > && callback );
+        virtual void register_option_to_update( rs2_option id, std::shared_ptr< option > option, bool update_from_hw = false );
```

Comment:
There's currently no use of this at all?

Reply:
Currently not for D400 devices.

Reply:
What criteria do you use to decide which options need to be registered?

Reply:
For now the one from [RSDEV-1025].
Later we can add more. like temperatures, and change the update mechanism used in the viewer.